client/components/data/query-tld-validation-schema/index.jsx

```diff
@@ -0,0 +1,52 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import { difference, isEmpty, isEqual, keys } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getValidationSchemas } from 'state/selectors';
+import { requestValidationSchemas } from 'state/domains/management/validation-schemas/actions';
+
+export class QueryTldValidationSchema extends Component {
+	static propTypes = {
+		validationSchemas: PropTypes.object.isRequired,
+		requestValidationSchemas: PropTypes.func.isRequired,
+		tlds: PropTypes.array.isRequired,
+	};
+
+	componentDidMount() {
+		this.fetchMissingSchemas( this.props.tlds );
+	}
+
+	componentWillReceiveProps( nextProps ) {
```

Comment:
fun fact!

if you destructure in the arguments you won't need a separate declaration for `newTlds`

```js
componentWillReceiveProps( { tlds } ) {
  if ( ! isEqual( this.props.tlds, tlds ) ) {
    this.fetchMissingSchemas( tlds );
  }
}
```

also as a note we probably would prefer to do this in `componentDidUpdate()` in order to keep side-effects out of `willMount` / `willReceiveProps` - I'm fuzzy on this but I believe @gwwar recently informed me that there can be troubles otherwise.

Reply:
Let's see current recs are to avoid `setState` in *Did to avoid unnecessary renders but it can be a good place for DOM manipulation/side-effects.
https://reactjs.org/docs/react-component.html#componentwillmount
https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9

I think dispatching redux actions eventually triggers a setState call, but I think we're still better off treating redux dispatches as a sid- effect since not all of our actions are idempotent. (eg componentWillMount in some cases might be called multiple times).

Reply:
Also I think that we’re looking at setState calls that are coming from the outside and probably after some appreciable time. That is, if it’s a Redux dispatch it will cause a re-render regardless if it’s in will or did

Reply:
> Also I think that we’re looking at setState calls that are coming from the outside and probably after some appreciable time. 

Yeah, `fetchMissingSchemas()` won't trigger any props changes until the request comes back, and no `setState()`s at all, so `componentDidUpdate()` seems the right place for it.

Thanks for the cheatsheet link! - I haven't seen that before, and I love it :D